torch/csrc/autograd/variable.h

```diff
@@ -348,34 +404,24 @@ struct TORCH_API Variable::Impl : public at::TensorImpl {
   /// Reset all expensive fields to free up resources
   void release_resources() override;
 
-  std::string name;
-  at::Tensor data_;
-
-  Variable grad_;
-  std::shared_ptr<Function> grad_fn_;
-  std::weak_ptr<Function> grad_accumulator_;
-
-  VariableVersion version_counter_;
-  std::vector<std::shared_ptr<FunctionPreHook>> hooks_;
-
-  // Only meaningful on leaf variables (must be false otherwise)
-  bool requires_grad_;
+  Variable::AutogradMeta* get_autograd_meta() const {
+    return static_cast<Variable::AutogradMeta*>(data_.unsafeGetTensorImpl()->autograd_meta());
+  }
 
-  bool is_view_;
+  int64_t storage_offset() const override;
 
-  // The "output number" of this variable; e.g., if this variable
-  // was the second output of a function, then output_nr == 1.
-  // We use this to make sure we can setup the backwards trace
-  // correctly when this variable is passed to another function.
-  uint32_t output_nr_;
-  PyObject* pyobj_; // weak reference
+  void set_allow_size_or_storage_change(bool value) override {
+    data_.unsafeGetTensorImpl()->set_allow_size_or_storage_change(value);
+  }
 
-  // Mutex to ensure that concurrent read operations that modify internal
-  // state are still thread-safe. Used by get_grad_fn and
-  // get_grad_accumulator.
-  std::mutex mutex_;
+  bool allow_size_or_storage_change() const override {
+    return data_.unsafeGetTensorImpl()->allow_size_or_storage_change();
+  }
 
-  int64_t storage_offset() const override;
+  bool defined() const override {
+    return data_.unsafeGetTensorImpl()->defined();
+  }
+  at::Tensor data_;
```

Comment:
nit: can you make this more prominent?  It's easy to miss because of the spacing.